README.md

```diff
@@ -6,6 +6,43 @@
 
 `supportedLanguages.json` is the shared list of all supported locales across FxA
 
+
+### oauth
+
+`oauth.scopes` provides shared logic for validating and checking OAuth scopes.
+
+Given a scope string, you can parse it into a `Scope` object either from a
+raw space-delimited string:
+
+```
+let s1 = oauth.scope.fromString('profile:write basket');
+```
+
+Or directly from a url-encoded string:
+
+```
+let s2 = oauth.scope.fromString('profile%3Aemail+profile%3Adisplay_name+clients');
+```
+
+Once you have a scope object,
+you can check whether
+it is sufficient to imply another scope:
+
+```
+  s1.implies(oauth.scope.fromString('profile:email:write')); // true
+  s2.implies(oauth.scope.fromString('profile:email:write')); // false
+```
+
+Or you can filter it down
+to only the scope values
+implied by another scope:
+
+```
+  let s3 = oauth.scope.fromString('profile:email clients:abcd'));
+  s3.filter(s1); // "profile:email"
```

Comment:
nit: are you sure you want to use `filter`, would it be confusing due to Array.filter? or is this the point of calling it filter?

Reply:
Indeed, it was named by analogy to `Array.filter`, because it does the same sort of thing - returns a subset of the values that match a given condition.  But if it seems confusing we can try to come up with something else.

Reply:
The `filter` operation is what would be required for the proposed API in https://github.com/mozilla/fxa-oauth-server/issues/381

Reply:
> it does the same sort of thing 

Sounds good to me, just wanted to check 👍